src/Icicle/Source/Checker/Constraint.hs

```diff
@@ -171,10 +171,13 @@ generateQ qq@(Query (c:_) _)
     -- >    : Aggregate a
     GroupFold _ k v x
      -> do  (x',sx)    <- generateX x
+
             retKey     <- Temporality TemporalityElement . TypeVar <$> fresh
             retVal     <- Temporality TemporalityElement . TypeVar <$> fresh
+            removeElementBinds
             (q',sq,t') <- withBind k retKey
                         $ withBind v retVal rest
+
```

Comment:
hmm damn, I think you'll actually need to reset the context after here.
stupid - this is my fault. do you want the honours of killing the state monad?